etna/records/templatetags/records_tags.py

```diff
@@ -10,12 +11,24 @@
 
 
 @register.simple_tag
-def record_url(record: Union[Record, Dict[str, Any]]) -> str:
+def record_url(
+    record: Union[Record, Dict[str, Any]], is_editorial: bool = False
+) -> str:
     """
     Return the URL for the provided `record` dict; which could either be a
     full/transformed result from the fetch() endpoint, OR a raw result from
     the search() endpoint.
     """
+    if is_editorial and settings.FEATURE_RECORD_LINKS_GO_TO_DISCOVERY:
+        if iaid := record.get("iaid"):
+            return f"https://discovery.nationalarchives.gov.uk/details/r/{iaid}"
+        if url := record.get("sourceUrl"):
+            return url
+        try:
+            # Assume `record` is an un-transformed search result
+            return record_url(record["_source"]["@template"]["details"], True)
```

Comment:
Does this work for record type?

Reply:
Try not to focus on this too much, as all I'm doing is indenting the original code here. It's not my intention to change any of this logic.

Reply:
Lets leave intentions out of this. I find it wired since different types seem to be handled here and therefore confusing. The original piece of work doe not have such logic anyways for handling this. This seems to be newly introduced.

Reply:
@TNA-Allan I've refactored to avoid repeating lines unnecessarily. Please mark as resolved if you are happy :)

Reply:
@ababic I think it tricky to understand and debug this method since it caters to 2 different kinds in instances for Record.

Reply:
@ababic I think it tricky to understand and debug this method since it caters to 2 different kinds in instances for Record.